### DISCUSSION TITLE
Transport equations in pores and throats without considering pore-throat-pore assembly

### DISCUSSION THREAD

**Discussion by harshpallam** [PMEAL/OpenPNM#2614] [Q&A]
Hi,
I am running transient transport simulations in OpenPNM. If I am correct, OpenPNM solves for concentrations at pores only by considering pore-throat-pore assembly. Also, in the examples given for OpenPNM, adsorption or reaction happens only in pores. However, in my model, I have separate equations for pores and throats, and sorption happens only in the throats (not in pores). Please find the attached file with equations for pores and throats.  Can you please suggest an example code where separate equations are implemented in pores and throats without considering pore-throat-pore assembly? Also, kindly let me know if there is a way to incorporate the sorption term for the throats.

I look forward to your response.
Thanks in advance.
![Screenshot from 2022-10-26 22-31-42](https://user-images.githubusercontent.com/95679531/198092030-09f5209a-3866-4bf9-bc8d-e5f8b64e45c0.png)

**Comment by ma-sadeghi** [comment]
Hi @harshpallam,

You can add a new node for each throat in the network, then you have two sets of governing equations, one for your pores and one for your throats. OpenPNM has built-in functions to do the network manipulation part, the transport part I think can be done but requires more thinking as it's now a research problem.

For starters, however, I suggest that you do a simplification: assume everything happens in the pores, so you don't need to deal with the extra coupling. What I mean is that, do your adsorption calculations using throat properties (surface area, etc.), but assign it to the adjacent pores. Specifically, if the rate of adsroption at throat $ij$ is $r_{ij}$, assign half of it to pore $i$ and the other half to pore $j$.

This can easily be done by writing a custom pore-scale model. OpenPNM already has a built-in pore-scale model for calculating the effective volume of pores by adding half of the volume of the connecting throats to the volume of each pore, which you can copy and modify to suit your needs:

https://github.com/PMEAL/OpenPNM/blob/53d7a159e393f3c051674bf7c9bf410bf79caf1f/openpnm/models/geometry/pore_volume/_funcs.py#L96-L122

Please let me know if this makes sense.

**Comment by harshpallam** [comment]
Dear @ma-sadeghi sir,
Thanks for the suggestion. I will think about it and let you know.
